extensions/coroutines-extensions/src/commonMain/kotlin/com/squareup/sqldelight/runtime/coroutines/FlowExtensions.kt

```diff
@@ -20,81 +20,69 @@ package com.squareup.sqldelight.runtime.coroutines
 
 import com.squareup.sqldelight.Query
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowViaChannel
-import kotlinx.coroutines.flow.flowWith
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.conflate
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.withContext
 import kotlin.coroutines.CoroutineContext
 import kotlin.jvm.JvmName
 import kotlin.jvm.JvmOverloads
 
 /** Turns this [Query] into a [Flow] which emits whenever the underlying result set changes. */
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmName("toFlow")
-fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = flowViaChannel(CONFLATED) { channel ->
+fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = callbackFlow<Query<T>> {
   val listener = object : Query.Listener {
     override fun queryResultsChanged() {
-      channel.offer(this@asFlow)
+      offer(this@asFlow)
     }
   }
   addListener(listener)
-  @Suppress("EXPERIMENTAL_API_USAGE") //TODO: Remove when invokeOnClose is no longer experimental, or use replacement.
-  channel.invokeOnClose {
+  awaitClose {
     removeListener(listener)
   }
-  channel.offer(this@asFlow)
-}
+  offer(this@asFlow)
+}.conflate()
 
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmOverloads
 fun <T : Any> Flow<Query<T>>.mapToOne(
   context: CoroutineContext = Dispatchers.Default
-): Flow<T> {
-  return flowWith(context) {
-    map { it.executeAsOne() }
+): Flow<T> = map {
+  withContext(context) {
+    it.executeAsOne()
   }
 }
 
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmOverloads
 fun <T : Any> Flow<Query<T>>.mapToOneOrDefault(
   defaultValue: T,
   context: CoroutineContext = Dispatchers.Default
-): Flow<T> {
-  return flowWith(context) {
-    map { it.executeAsOneOrNull() ?: defaultValue }
-  }
+): Flow<T> = map {
+  withContext(context) { it.executeAsOneOrNull() } ?: defaultValue
 }
 
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmOverloads
 fun <T : Any> Flow<Query<T>>.mapToOneOrNull(
   context: CoroutineContext = Dispatchers.Default
-): Flow<T?> {
-  return flowWith(context) {
-    map { it.executeAsOneOrNull() }
-  }
-}
+): Flow<T?> = map { withContext(context) { it.executeAsOneOrNull() } }
 
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmOverloads
+//TODO: Rename to mapToOneNotNull for consistency with Kotlin stdlib.
```

Comment:
Unsure of the importance of this. There's also consistency in the library.

Reply:
Currently, this is only used in RxJava extensions for released API.
RxJava API is not consistent with Flow and Kotlin APIs, so I think it's better to be closer to Kotlin convention for that new API that is almost Kotlin-only as suspending functions, a "primitive" that `Flow` relies on, don't work in Java.

Do you want me to make a separate PR for that? I can provide a deprecated overload for the "NonNull" naming (with a `replaceWith` clause) to guide any users migrating from RxJava, much like `Flow` has deprecated extension functions that target users familiar with RxJava (e.g. `subscribe`, `observeOn`).

Reply:
Just rename it. No need to deprecate.